IOSExercises/Views/ProfilesTableViewController.m

```diff
@@ -61,39 +69,105 @@ - (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(N
-                                           color:[EXRColor supportedColor:EXRSupportedColorRed]];
-
-    self.persons = @[latte, mocha];
+    self.persons = [Person fetchAllPersonsWithContext:self.dataController.managedObjectContext];
+}
+
+- (NSArray<Person *> *)fetchPersonsWithBirthdayToday {
+
+    NSArray<Person *> *persons = [Person fetchAllPersonsWithContext:self.dataController.managedObjectContext];
+
+    NSPredicate *birthdayPredicate = [NSPredicate predicateWithBlock:^BOOL(id  _Nullable evaluatedObject, NSDictionary<NSString *,id> * _Nullable bindings) {
+
+        Person *fetchedPerson = (Person *) evaluatedObject;
+
+        return [self isBirthdayTodayOfPerson:fetchedPerson];
+    }];
+    return [persons filteredArrayUsingPredicate:birthdayPredicate];
+}
+
+- (BOOL)isBirthdayTodayOfPerson:(Person *)person {
+    NSDateComponents *birthdayComponent = [[NSCalendar currentCalendar] components:NSCalendarUnitDay | NSCalendarUnitMonth
+                                                                          fromDate:person.birthday];
+    NSDateComponents *todayComponent = [[NSCalendar currentCalendar] components:NSCalendarUnitDay | NSCalendarUnitMonth
+                                                                       fromDate:[NSDate date]];
+    return birthdayComponent.day == todayComponent.day
+        && birthdayComponent.month == todayComponent.month;
+}
+
+
+#pragma mark - Action Handler Methods
+
+- (IBAction)didTapFilterButton:(id)sender {
+    UIButton *filterButton = (UIButton *)sender;
+    NSString *buttonTitle = filterButton.titleLabel.text;
+    NSLog(@"ProfilesTableViewController: %@ Filter Button was tapped", buttonTitle);
+    if ([buttonTitle isEqualToString:@"All"]) {
+        [self setupPersons];
+    } else {
+        _persons = [Person fetchPersonsWithColor:[EXRColor colorWithName:buttonTitle]
+                                     withContext:self.dataController.managedObjectContext];
+    }
+    self.birthdayListSwitch.on = NO;
+    [self.tableView reloadData];
+}
+
+- (void) switchIsChanged:(UISwitch *)paramSender{
+    NSLog(@"ProfilesTableViewController: Switch is tapped");
+    if ([paramSender isOn]){
+        _persons = [self fetchPersonsWithBirthdayToday];
+        NSLog(@"ProfilesTableViewController: Switch is on");
+    } else {
+        [self setupPersons];
+        NSLog(@"ProfilesTableViewController: Switch is off");
+    }
+    [self.tableView reloadData];
+}
+
+#pragma mark - Layout Helper Methods
+
+- (void)setupCellHeight {
+    self.tableView.estimatedRowHeight = 44.f;
+    self.tableView.rowHeight = UITableViewAutomaticDimension;
+}
+
+- (UIBarButtonItem *)createBarButtonItemWithTitle:(NSString *)aTitle {
+    UIButton *barButton = [UIButton buttonWithType:UIButtonTypeSystem];
+    [barButton setTitle:aTitle
+               forState:UIControlStateNormal];
+    [barButton addTarget:self
+                  action:@selector(didTapFilterButton:)
```

Comment:
This method specifically creates a button for filtering, so might need to reflect method name for this